odl/trafos/backends/pyfftw_bindings.py

```diff
@@ -0,0 +1,313 @@
+    ----------------
+    fftw_plan : ``pyfftw.FFTW``, optional
+        Use this plan instead of calculating a new one. If specified,
+        the options ``planning_effort``, ``planning_timelimit`` and
+        ``threads`` have no effect.
+    planning_effort : {'estimate', 'measure', 'patient', 'exhaustive'}
+        Flag for the amount of effort put into finding an optimal
+        FFTW plan. See the `FFTW doc on planner flags
+        <http://www.fftw.org/fftw3_doc/Planner-Flags.html>`_.
+        Default: 'estimate'
+    planning_timelimit : float or ``None``, optional
+        Limit planning time to roughly this many seconds.
+        Default: ``None`` (no limit)
+    threads : int, optional
+        Number of threads to use.
+        Default: Number of CPUs if the number of data points is larger
+        than 4096, else 1.
+    normalise_idft : bool, optional
+        If ``True``, the result of the backward transform is divided by
+        ``1 / N``, where ``N`` is the total number of points in
+        ``array_in[axes]``. This ensures that the IDFT is the true
+        inverse of the forward DFT.
+        Default: ``False``
+    import_wisdom : filename or file handle, optional
+        File to load FFTW wisdom from. If the file does not exist,
+        it is ignored.
+    export_wisdom : filename or file handle, optional
+        File to append the accumulated FFTW wisdom to
+
+    Returns
+    -------
+    fftw_plan : ``pyfftw.FFTW``
+        The plan object created from the input arguments. It can be
+        reused for transforms of the same size with the same data types.
+        Note that reuse only gives a speedup if the initial plan
+        used a planner flag other than ``'estimate'``.
+        If ``fftw_plan`` was specified, the returned object is a
+        reference to it.
+
+    Notes
+    -----
+    * The planning and direction flags can also be specified as
+      capitalized and prepended by ``'FFTW_'``, i.e. in the original
+      FFTW form.
+    * For a ``halfcomplex`` forward transform, the arrays must fulfill
+      ``array_out.shape[axes[-1]] == array_in.shape[axes[-1]] // 2 + 1``,
+      and vice versa for backward transforms.
+    * All planning schemes except ``'estimate'`` require an internal copy
+      of the input array but are often several times faster after the
+      first call (measuring results are cached). Typically,
+      'measure' is a good compromise. If you cannot afford the copy,
+      use ``'estimate'``.
+    * If a plan is provided via the ``fftw_plan`` parameter, no copy
+      is needed internally.
+    """
+    import pickle
+
+    if not array_in.flags.aligned:
+        raise ValueError('input array not aligned')
+
+    if not array_out.flags.aligned:
+        raise ValueError('output array not aligned')
+
+    if axes is None:
+        axes = tuple(range(array_in.ndim))
+
+    axes = normalized_axes_tuple(axes, array_in.ndim)
+
+    direction = _pyfftw_to_local(direction)
```

Comment:
I know. But this was a pure code move, and there is a separate issue #616 on that. Otherwise this branch never goes in.